### PR TITLE
Include bounds in generic re-ordering diagnostic

### LINKED ISSUE #59508: Generic parameter reordering diagnostic doesn't contain bounds
```rust
struct A;

impl A {
    pub fn do_things<T, 'a, 'b: 'a>() {
        println!("panic");
    }
}
```
produces:
```
error: lifetime parameters must be declared prior to type parameters
 --> src/lib.rs:4:25
  |
4 |     pub fn do_things<T, 'a, 'b: 'a>() {
  |                     ----^^--^^----- help: reorder the parameters: lifetimes, then types, then consts: `<'a, 'b, T>`
```
which should be:
```
error: lifetime parameters must be declared prior to type parameters
 --> src/lib.rs:4:25
  |
4 |     pub fn do_things<T, 'a, 'b: 'a>() {
  |                     ----^^--^^----- help: reorder the parameters: lifetimes, then types, then consts: `<'a, 'b: 'a, T>`
```

### PR DESCRIPTION
**Pull request by davidtwco**
Fixes #59508.

r? @estebank
cc @varkor

### CODE REVIEW

File: src/test/ui/issue-59508-1.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#![allow(dead_code)]
+#![feature(const_generics)]
+//~^ WARN the feature `const_generics` is incomplete and may cause the compiler to crash
+
+// This test checks that generic parameter re-ordering diagnostic suggestsions mention that
```
Comment (varkor) [inline src/test/ui/issue-59508-1.rs]:
```suggestion
// This test checks that generic parameter re-ordering diagnostic suggestions mention that
```

File: src/test/ui/issue-59508.fixed [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// run-rustfix
+
+#![allow(dead_code)]
+
+// This test checks that generic parameter re-ordering diagnostic suggestsions contain bounds.
```
Comment (varkor) [inline src/test/ui/issue-59508.fixed]:
```suggestion
// This test checks that generic parameter re-ordering diagnostic suggestions contain bounds.
```

File: src/test/ui/issue-59508.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// run-rustfix
+
+#![allow(dead_code)]
+
+// This test checks that generic parameter re-ordering diagnostic suggestsions contain bounds.
```
Comment (varkor) [inline src/test/ui/issue-59508.rs]:
```suggestion
// This test checks that generic parameter re-ordering diagnostic suggestions contain bounds.
```

**Review by varkor — APPROVED** [review]
Thanks! r=me with typos fixed.

**Comment by Centril** [comment]
Rolled up into https://github.com/rust-lang/rust/pull/59579, @bors retry

**Pull request merged**